gobblin-service/src/main/java/org/apache/gobblin/service/modules/spec/JobExecutionPlan.java

```diff
@@ -65,6 +65,7 @@
   private final SpecExecutor specExecutor;
   private ExecutionStatus executionStatus = ExecutionStatus.PENDING;
   private final int maxAttempts;
+  private int currentGeneration = 1;
   private int currentAttempts = 0;
```

Comment:
above, it looked like "1" is the default/fallback for both generation and attempts, but here and below it seems that attempts begins at 0, but generation begins as 1.
1. why a different starting value for each?
2. for whatever starting value we use, is there a discrepancy between the `getProp` fallback and here?

Reply:
1. we only increase the number of current generation when we send pending resume, and for currentAttempts, we increase it every time we call orchestrated. So for the first attempt, once we orchestrate it, the current attempt and current generation will all be 1. Before first job orchestrated, when the job status is pending, it will be 1 generation and 0 attempts. 
2. No, both of the default value is 1